contrib/xz-cmake/CMakeLists.txt

```diff
@@ -0,0 +1,246 @@
+        ${LIBRARY_DIR}/liblzma/api/lzma/stream_flags.h
+        ${LIBRARY_DIR}/liblzma/api/lzma/version.h
+        ${LIBRARY_DIR}/liblzma/api/lzma/vli.h
+        ${LIBRARY_DIR}/liblzma/check/check.h
+        ${LIBRARY_DIR}/liblzma/check/crc32_table_be.h
+        ${LIBRARY_DIR}/liblzma/check/crc32_table_le.h
+        ${LIBRARY_DIR}/liblzma/check/crc64_table_be.h
+        ${LIBRARY_DIR}/liblzma/check/crc64_table_le.h
+        ${LIBRARY_DIR}/liblzma/check/crc_macros.h
+        ${LIBRARY_DIR}/liblzma/common/alone_decoder.h
+        ${LIBRARY_DIR}/liblzma/common/block_buffer_encoder.h
+        ${LIBRARY_DIR}/liblzma/common/block_decoder.h
+        ${LIBRARY_DIR}/liblzma/common/block_encoder.h
+        ${LIBRARY_DIR}/liblzma/common/common.h
+        ${LIBRARY_DIR}/liblzma/common/easy_preset.h
+        ${LIBRARY_DIR}/liblzma/common/filter_common.h
+        ${LIBRARY_DIR}/liblzma/common/filter_decoder.h
+        ${LIBRARY_DIR}/liblzma/common/filter_encoder.h
+        ${LIBRARY_DIR}/liblzma/common/index_decoder.h
+        ${LIBRARY_DIR}/liblzma/common/index_encoder.h
+        ${LIBRARY_DIR}/liblzma/common/index.h
+        ${LIBRARY_DIR}/liblzma/common/memcmplen.h
+        ${LIBRARY_DIR}/liblzma/common/outqueue.h
+        ${LIBRARY_DIR}/liblzma/common/stream_decoder.h
+        ${LIBRARY_DIR}/liblzma/common/stream_flags_common.h
+        ${LIBRARY_DIR}/liblzma/delta/delta_common.h
+        ${LIBRARY_DIR}/liblzma/delta/delta_decoder.h
+        ${LIBRARY_DIR}/liblzma/delta/delta_encoder.h
+        ${LIBRARY_DIR}/liblzma/delta/delta_private.h
+        ${LIBRARY_DIR}/liblzma/lz/lz_decoder.h
+        ${LIBRARY_DIR}/liblzma/lz/lz_encoder.h
+        ${LIBRARY_DIR}/liblzma/lz/lz_encoder_hash.h
+        ${LIBRARY_DIR}/liblzma/lz/lz_encoder_hash_table.h
+        ${LIBRARY_DIR}/liblzma/lzma/fastpos.h
+        ${LIBRARY_DIR}/liblzma/lzma/lzma2_decoder.h
+        ${LIBRARY_DIR}/liblzma/lzma/lzma2_encoder.h
+        ${LIBRARY_DIR}/liblzma/lzma/lzma_common.h
+        ${LIBRARY_DIR}/liblzma/lzma/lzma_decoder.h
+        ${LIBRARY_DIR}/liblzma/lzma/lzma_encoder.h
+        ${LIBRARY_DIR}/liblzma/lzma/lzma_encoder_private.h
+        ${LIBRARY_DIR}/liblzma/rangecoder/price.h
+        ${LIBRARY_DIR}/liblzma/rangecoder/range_common.h
+        ${LIBRARY_DIR}/liblzma/rangecoder/range_decoder.h
+        ${LIBRARY_DIR}/liblzma/rangecoder/range_encoder.h
+        ${LIBRARY_DIR}/liblzma/simple/simple_coder.h
+        ${LIBRARY_DIR}/liblzma/simple/simple_decoder.h
+        ${LIBRARY_DIR}/liblzma/simple/simple_encoder.h
+        ${LIBRARY_DIR}/liblzma/simple/simple_private.h
+)
+
+ADD_LIBRARY(liblzma ${Sources} ${Headers})
+
+target_include_directories(liblzma PUBLIC
+    ${LIBRARY_DIR}/liblzma/api
+    ${LIBRARY_DIR}/liblzma/common
+    ${LIBRARY_DIR}/liblzma/check
+    ${LIBRARY_DIR}/liblzma/lz
+    ${LIBRARY_DIR}/liblzma/rangecoder
+    ${LIBRARY_DIR}/liblzma/lzma
+    ${LIBRARY_DIR}/liblzma/delta
+    ${LIBRARY_DIR}/liblzma/simple
+    ${LIBRARY_DIR}/common
+)
```

Comment:
Does all of this should be public includes? (i.e. the one that will be added while building library itself and those who linked against it)

Reply:
Completely removed extra folder for CMake, exposed for includes only one folder `liblzma/api`